### PR TITLE
small fix to the failing tests

### PR DESCRIPTION
**Pull request by djarecka**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Summary

ok, I added a very simple fix. And I will work on a separate PR

## Checklist

- [ ] All tests passing
- [ ] I have added tests to cover my changes
- [ ] I have updated documentation (if necessary)
- [x] My code follows the code style of this project 
(we are using `black`: you can `pip install pre-commit`, 
run `pre-commit install` in the `pydra` directory 
and `black` will be run automatically with each commit)

## Acknowledgment
- [x] I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

File: pydra/engine/tests/test_specs.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def test_file_hash(tmpdir):
     inputs = make_klass(input_spec)
     assert (
         inputs(in_file=outfile).hash
-        == "1384a1eb11cd94a5b826a82b948313b9237a0956d406ccff59e79ec92b3c935f"
+        == "d4dc839034f90419ce528e992f5bf9cecf876da14d81ec5a19d0d03271d90e02"
```
Comment (satra) [inline pydra/engine/tests/test_specs.py]: why is this hash changing?

File: pydra/engine/specs.py [hunk 2]
```diff
@@ -39,13 +39,13 @@ def collect_additional_outputs(self, input_spec, inputs, output_dir):
...
         """Compute a basic hash for any given set of fields."""
-        from .helpers import hash_function
+        from .helpers import hash_function, ensure_list
         from .helpers_file import hash_file
 
         inp_dict = {
             field.name: hash_file(getattr(self, field.name))
             if (field.type == File and "container_path" not in field.metadata)
-            else getattr(self, field.name)
+            else ensure_list(getattr(self, field.name), tuple2list=True)
```
Comment (satra) [inline pydra/engine/specs.py]: this is turning every input into a list.

Comment (djarecka) [reply pydra/engine/specs.py]: i've changed it

File: pydra/engine/tests/test_specs.py (hunk 1 above)
Comment (djarecka) [reply pydra/engine/tests/test_specs.py]: you were right, was not needed...

File: pydra/engine/specs.py [hunk 3]
```diff
@@ -39,19 +39,21 @@ def collect_additional_outputs(self, input_spec, inputs, output_dir):
     @property
     def hash(self):
         """Compute a basic hash for any given set of fields."""
-        from .helpers import hash_function
+        from .helpers import hash_function, ensure_list
```
Comment (satra) [inline pydra/engine/specs.py]:
```suggestion
        from .helpers import hash_function
```

File: pydra/engine/specs.py [hunk 4]
```diff
@@ -39,19 +39,21 @@ def collect_additional_outputs(self, input_spec, inputs, output_dir):
...
-        }
+        inp_dict = {}
+        for field in attr_fields(self):
+            if field.name not in [
+                "_graph_checksums",
+                "bindings",
+            ] and not field.metadata.get("output_file_template"):
+                if field.type == File and "container_path" not in field.metadata:
+                    inp_dict[field.name] = hash_file(getattr(self, field.name))
+                elif isinstance(getattr(self, field.name), tuple):
```
Comment (satra) [inline pydra/engine/specs.py]: just a note that we should address List/tuples of Files for hashing or dictionaries. i.e. nested input fields

Comment (djarecka) [reply pydra/engine/specs.py]: https://github.com/nipype/pydra/issues/178

**Pull request merged**